src/mlagility/cli/report.py

```diff
@@ -1,521 +1,204 @@
-    printing.logn(str(results_file), printing.Colors.OKGREEN)
+    # Description: Number of chips needed to run the model when mapped to Groq
+    # Source: Computed by GroqFlow
+    groq_chips_used: str = "-"
 
+    # Description: Number of chips needed to run the model when mapped to Groq
+    # Source: Computed by GroqFlow
+    groq_chips_used: str = "-"
 
-def op_spreadsheet(cache_dir):
+    # Description: An estimation of latency, in ms, to run one invocation of the
+    #              model on a GroqNode server (includes I/O).
+    # Source: computed by GroqFlow
+    groq_estimated_latency: str = "-"
 
-    # Declare file paths
-    spreadsheet_file = os.path.join(cache_dir, op_support_filename)
+    # Description: Mean total latency, in ms, to run one invocation using TensorRT.
+    # Source: benchit.benchmark() on 100 runs
+    nvidia_latency: str = "-"
 
-    with open(spreadsheet_file, "w", newline="", encoding="utf8") as spreadsheet:
-        writer = csv.writer(spreadsheet)
+    # Description: Mean total latency, in ms, to run one invocation using OnnxRuntime.
+    # Source: benchit.benchmark() on 100 runs
+    x86_latency: str = "-"
+
+
+def summary_spreadsheet(args) -> None:
+
+    # Input arguments from CLI
+    cache_dirs = [os.path.expanduser(dir) for dir in args.cache_dirs]
+    report_dir = os.path.expanduser(args.report_dir)
 
-        # Git commit placeholder
-        writer.writerow(
-            [
-                "Git Commit: ",
-            ]
-        )
+    # Name report file
+    report_path = os.path.join(report_dir, get_report_name())
 
-        writer.writerow(
-            [
-                "Model Name",
-                "Unsupported ops (Optimized ONNX)",
-            ]
-        )
+    # Create report dict
+    Path(report_dir).mkdir(parents=True, exist_ok=True)
+    report = {}
 
+    # Add results from all cache folders
+    for cache_dir in cache_dirs:
+
+        # List all yaml files available
         all_model_state_yamls = cache.get_all(path=cache_dir, file_type="state.yaml")
         all_model_state_yamls = sorted(all_model_state_yamls)
 
+        # Add each model to report
         for model_state_yaml in all_model_state_yamls:
+
+            # Models are identified my the build name
+            build_name = model_state_yaml.split("/")[-2]
+
+            # Load state
             state = build.load_state(state_path=model_state_yaml)
 
-            writer.writerow(
-                [
-                    state.config.build_name,  # Model Name
-                    _cleanOpList(
-                        state.info.opt_onnx_unsupported_ops
-                    ),  # Unsupported ops (Optimized ONNX)
-                ]
+            # Add model to report if it doesn't exist
+            if build_name not in report:
+                report[build_name] = BuildResults()
+
+            # Get model hash from build name
+            report[build_name].model_hash = state.config.build_name.split("_")[-1]
```

Comment:
Let's avoid these magic formulas. Put `model_hash` into `State.Info` during the build instead?

Reply:
Simplified to `build_name.split("_")[-1]` so it looks less ominous.
Also, it looks like `model_hash` is being saved as part of state, but it doesn't match the model hash that is part of the build name. I created an issue to address that.